core/trino-main/src/main/java/io/trino/server/HttpRequestSessionContextFactory.java

```diff
@@ -254,13 +254,19 @@ private Identity buildSessionIdentity(Optional<Identity> authenticatedIdentity,
         String user = trinoUser != null ? trinoUser : authenticatedIdentity.map(Identity::getUser).orElse(null);
```

Comment:
I would add to the commit message. 

> That the previous commit broke the contract between engine and plugins, and it was no longer passing desired role to be enabled. And that this commit restores that. The change has no impact to the project but affects 3rd party plugins that relied on the previous behavior.

Please do not copy paste blindly, I wanted to give you content, but form most likely requires some polishing. 

Please also update PR titlte and description to match that.